src/mongo/tree/MongoCollectionTreeItem.ts

```diff
@@ -165,9 +165,9 @@ export class MongoCollectionTreeItem implements IAzureParentTreeItem {
 			throw new Error("Too many arguments")
 		}
 		let result;
-		if (args.length === 1) {
+		if (args && args.length === 1) {
```

Comment:
You shouldn't have to check 'args' for every 'else' case if you check it in the first 'if'. Something like this would be less bug-prone IMO:
```
if (!args || args.length === 0) {

} else if (args.length === 1) {

} else if (args.length === 2) {

} else {

}
```